### ISSUE TITLE
IDE0043 error is incorrect when using literal array

### ISSUE THREAD

**Issue by jrmoreno1** [dotnet/roslyn#42764]
**Version Used**: 16.5.1

**Steps to Reproduce**:

```vb
Sub main()
    System.Console.WriteLine(String.Format("{0} {1}", {"hello", "world"}))
End Sub
```

**Expected Behavior**: No warnings

**Actual Behavior**: Warning	IDE0043	Format string contains invalid placeholder
